src/transaction/guards.ts

```diff
@@ -32,12 +36,33 @@ export const isTxInput = (
   return Buffer.isBuffer(txHash) && Number.isInteger(outputIndex)
 }
 
+// export const isAsset = (
+//   test: any,
+// ): test is Asset => test instanceof Map
+//   && Object.entries(test).every(([assetName, amount]) => Buffer.isBuffer(assetName) && isBigNumber(amount))
+
+// export const isMultiAsset = (
+//   test: any,
+// ): test is MultiAsset => test instanceof Map
+//   && Object.entries(test).every(([policyId, asset]) => Buffer.isBuffer(policyId) && isAsset(asset))
+
 export const isTxOutput = (
   test: any,
 ): test is TxOutput => {
-  if (!Array.isArray(test) || test.length !== 2) return false
-  const [address, coins] = test
-  return Buffer.isBuffer(address) && isLovelace(coins)
+  if (Array.isArray(test) && test.length === 2) {
+    const [address, coinsOrMA] = test
+    if (Buffer.isBuffer(address)) {
+      return true
+      // if (isLovelace(coinsOrMA)) {
+      //   return true
+      // }
+      // if (Array.isArray(coinsOrMA) && coinsOrMA.length === 2) {
+      //   const [coins, multiAsset] = coinsOrMA
+      //   return isLovelace(coins) && isMultiAsset(multiAsset)
+      // }
+    }
+  }
+  return false
```

Comment:
I initially wanted to write guards for whole `TxOutput`, but making a whole structure check and returning `true/false` seems to me like we are throwing away a lot of information that can be used to give more specific errors.

Check `parseTxOutputs` function, which then calls `parseMultiAsset` and `parseAsset`. Structure is checked during iteration and when error occurs, we can provide more descriptive error.

@PeterBenc what do you think?